### PR TITLE
doc: Add documentation for EITS

### PR DESCRIPTION
**Pull request by Vge0rge**
-This adds a documentation document for the
 encrypted ITS in TF-M.

Ref: NCSDK-7599

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

### CODE REVIEW

File: docs/technical_references/design_docs/nordic_eits.rst [hunk 1]
```diff
@@ -0,0 +1,73 @@
+====================================================
+Encryption in Internal Trusted Storage (ITS) Service
+====================================================
+
+:Author: TBD
+:Organization: TBD
+:Contact: TBD
```
Comment (joerchan) [inline docs/technical_references/design_docs/nordic_eits.rst:7]: When do we decide this?

File: docs/technical_references/design_docs/nordic_eits.rst [hunk 2]
```diff
@@ -0,0 +1,73 @@
+====================================================
+Encryption in Internal Trusted Storage (ITS) Service
```
Comment (joerchan) [inline docs/technical_references/design_docs/nordic_eits.rst:2]: Isn't the name "Encrypted Internal Trusted Storage (EITS)"?

File: docs/technical_references/design_docs/nordic_eits.rst [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+====================================================
+
+:Author: TBD
+:Organization: TBD
+:Contact: TBD
+
+Introduction
+============
```
Comment (joerchan) [inline docs/technical_references/design_docs/nordic_eits.rst:10]: blank line after this?

File: docs/technical_references/design_docs/nordic_eits.rst [hunk 4]
```diff
@@ -0,0 +1,73 @@
+====================================================
+Encryption in Internal Trusted Storage (ITS) Service
+====================================================
+
+:Author: TBD
+:Organization: TBD
+:Contact: TBD
+
+Introduction
```
Comment (joerchan) [inline docs/technical_references/design_docs/nordic_eits.rst:9]: The introduction seems more argumentative than introductive. Would it make sense to have something call Justification, and leave the introduction to simply introduce the feature?

**Comment by joerchan** [comment]
@Vge0rge Should this go through technical documentation review?

File: docs/technical_references/design_docs/nordic_eits.rst [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+is used with a unique derivation label for each file. The input key of the KDF
+is the MKEK, a symmetric key stored in the Key Managment Unit (KMU) of the
+Nordic devices. The MKEK is protected by the KMU peripheral and its key material
+cannot be read by software, it can only be used by reference.
+
+The key derivation process generates the 256 bit symmetric encryption key which
+is used with the ChaChaPoly1305 authenticated encryption cipher. In order to
+strengthen our data integrity guarantees, the metadata of the ITS file (creation
+flags/size) are used as authenticated data in the encryption process. In our
+scheme we use a unique nonce for each encryption process. The nonce is generated
```
Comment (SebastianBoe) [inline docs/technical_references/design_docs/nordic_eits.rst:47]:
```suggestion
scheme we use a statistically unique nonce for each encryption process. The nonce is generated
```

File: docs/technical_references/design_docs/nordic_eits.rst [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+cipher with optional authenticated data. The cipher has hardware acceleration
+support for both of the TF-M enabled Nordic devices (nRF9160/nRF5340). The key
+size is 256 bits and each ITS file is encrypted with a different key.
+
+The first step of the process is the generation of the encryption key. To ensure
+a different key for every ITS file a key derivation function (KDF) based on CMAC
+is used with a unique derivation label for each file. The input key of the KDF
+is the MKEK, a symmetric key stored in the Key Managment Unit (KMU) of the
+Nordic devices. The MKEK is protected by the KMU peripheral and its key material
+cannot be read by software, it can only be used by reference.
```
Comment (SebastianBoe) [inline docs/technical_references/design_docs/nordic_eits.rst:41]: We should clarify that we use a different encryption key for each client_id+UUID file identifier.
Not for each file.

So key re-use is possible.

File: docs/technical_references/design_docs/nordic_eits.rst [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+The key derivation process generates the 256 bit symmetric encryption key which
+is used with the ChaChaPoly1305 authenticated encryption cipher. In order to
+strengthen our data integrity guarantees, the metadata of the ITS file (creation
+flags/size) are used as authenticated data in the encryption process. In our
+scheme we use a unique nonce for each encryption process. The nonce is generated
+by concatenating a random seed and an increasing counter. The random seed is
```
Comment (SebastianBoe) [inline docs/technical_references/design_docs/nordic_eits.rst:48]:
```suggestion
by concatenating an 8-byte random seed and a 4-byte increasing counter. The random seed is
```

**Pull request closed without merging**